### PR TITLE
CL2: Add isFatal param to WaitForPods and enable failing measurements

### PR DESCRIPTION
**Pull request by learnitall**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR introduces a set of commits that enable measurements to fail during their `Execute` method. Errors that are returned during a measurement's `Execute` method are passed up the stack to the simple test executor, which then passes them through a function called `isErrsCritical`. This function can control if the entire test should bail, based on the criticalness of the errors passed to it. Right now, it's a no-op:

https://github.com/kubernetes/perf-tests/blob/e7bdd15c9122e55d7dc267768190addc8fdf7b9e/clusterloader2/pkg/test/simple_test_executor.go#L386-L389

This PR modifies the function to check if any of the errors in the given list are fatal. A measurement can mark an error as fatal if it returns an error wrapped by a newly added `FatalError` type.

This PR also modifies the `WaitForPods` measurement to utilize this functionality. A new param named `isFatal` can control if a test should bail if the `WaitForPods` measurement is unable to successfully determine that a set of required pods are running and ready. This can be helpful for tests that require some infrastructure within the cluster to run. For example:

```yaml
name: is-fatal-test
namespace:
  number: 1
tuningSets:
- name: Uniform1qps
  qpsLoad:
    qps: 1
steps:
- name: Create test pod
  phases:
  - namespaceRange:
      min: 1
      max: 1
    replicasPerNamespace: 1
    tuningSet: Uniform1qps
    objectBundle:
    - basename: my-test-pod
      objectTemplatePath: ./my-test-pod.yaml
- name: Wait for test pod
  measurements:
  - Identifier: WaitForRunningPods
    Method: WaitForRunningPods
    Params:
      labelSelector: name=my-test-pod
      desiredPodCount: 1
      timeout: 60s
      isFatal: true
- name: Sleep
  measurements:
  - Identifier: Sleep
    Method: Sleep
    Params:
      duration: 10s
```

In this config, the `Sleep` measurement will only be executed if the test pod has been successfully deployed.

### CODE REVIEW

**Comment by learnitall** [comment]
cc: @marseel

**Comment by learnitall** [comment]
Gentle ping @mborsz and @wojtek-t. Do you mind taking a look when you have a moment? :pray:

File: clusterloader2/pkg/errors/fatal_error.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+*/
+
+package errors
+
+import (
+	"errors"
+	"fmt"
+)
+
+var ErrFatal = errors.New("fatal error")
```
Comment (marseel) [inline clusterloader2/pkg/errors/fatal_error.go:24]: As we already have `isErrsCritical` let's call it `ErrCritical` as well.
Or rename `isErrsCritical` to `isErrsFatal`

File: clusterloader2/pkg/measurement/common/wait_for_pods.go [hunk 2]
```diff
@@ -72,7 +79,16 @@ func (w *waitForRunningPodsMeasurement) Execute(config *measurement.Config) ([]m
 	if err != nil {
 		return nil, err
 	}
+
 	_, err = measurementutil.WaitForPods(ctx, podStore, options)
+	if err != nil {
```
Comment (marseel) [inline clusterloader2/pkg/measurement/common/wait_for_pods.go:84]: nit:
how about something simpler:
```
if err != nil && isFatal {
    return nil, errors.NewErrFatal(err)
}
```

**Review by marseel — COMMENTED** [review]
Looks good, just small nits.

File: clusterloader2/pkg/measurement/common/wait_for_pods.go (hunk 2 above)
Comment (learnitall) [reply clusterloader2/pkg/measurement/common/wait_for_pods.go:84]: Ah that's much better, thanks.

File: clusterloader2/pkg/errors/fatal_error.go (hunk 1 above)
Comment (learnitall) [reply clusterloader2/pkg/errors/fatal_error.go:24]: Good idea, I replaced `fatal` with `critical`

**Comment by learnitall** [comment]
Thanks for the review @marseel! I just pushed some changes addressing your feedback.

**Comment by marseel** [comment]
Thanks!
/lgtm
/approve

**Pull request merged**